test/parallel/test-single-executable-application.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+require('../common');
+
+// This tests the creation of a single executable application.
+
+const tmpdir = require('../common/tmpdir');
+const { accessSync, writeFileSync } = require('fs');
+const { execSync } = require('child_process');
+const { join } = require('path');
+const { match, strictEqual } = require('assert');
+
+const inputFile = join(tmpdir.path, 'sea.js');
+const requirableFile = join(tmpdir.path, 'requirable.js');
+const outputFile = join(tmpdir.path, process.platform === 'win32' ? 'sea.exe' : 'sea');
+
+tmpdir.refresh();
+
+writeFileSync(requirableFile, `
+module.exports = {
+  hello: 'world',
+};
+`);
+
+writeFileSync(inputFile, `
+require('../common');
+
+const { deepStrictEqual, strictEqual } = require('assert');
+const { dirname } = require('path');
+
+deepStrictEqual(process.argv, [process.execPath, process.execPath, '-a', '--b=c', 'd']);
+
+strictEqual(__filename, process.execPath);
+strictEqual(__dirname, dirname(process.execPath));
+strictEqual(module.exports, exports);
+strictEqual(require.main, module);
+
+const requirable = require('./requirable.js');
+deepStrictEqual(requirable, {
+  hello: 'world',
+});
+
+console.log('Hello, world!');
+`);
+const compileOutput = execSync(`${process.execPath} --compile=${outputFile} ${inputFile} 2>&1`);
+match(compileOutput.toString(), /--compile is an experimental feature/);
+accessSync(outputFile);
+
+// Verifying code signing using a self-signed certificate.
+if (process.platform === 'darwin') {
+  let codesignFound = false;
+  try {
+    execSync('command -v codesign');
+    codesignFound = true;
+  } catch (err) {
+    console.log(err.message);
+  }
+  if (codesignFound) {
+    execSync(`codesign --sign - ${outputFile}`);
+    execSync(`codesign --verify ${outputFile}`);
+  }
+} else if (process.platform === 'win32') {
+  let signtoolFound = false;
+  try {
+    execSync('where signtool');
+    signtoolFound = true;
+  } catch (err) {
+    console.log(err.message);
+  }
+  if (signtoolFound) {
+    execSync(`signtool sign /fd SHA256 ${outputFile}`);
+    execSync(`signtool verify /pa SHA256 ${outputFile}`);
+  }
+}
+
+const singleExecutableApplicationOutput = execSync(`${outputFile} -a --b=c d`);
```

Comment:
Looks like we have memory addressing issues here 🤔

Reply:
Hmm, I wasn't able to repro this locally on my macOS even with an ASAN build. We'll probably have to skip if if `process.config.variables.asan` is `true`.